### PR TITLE
作业

### PR DESCRIPTION
**Pull request by tianlinzhong**
on put

### CODE REVIEW

File: meter.py [hunk 1]
```diff
@@ -168,18 +168,49 @@ def on_put(req, resp, id_):
...
+                                   description='API.METER_ID_IS_ALREADY_IN_USE')
         # TODO 4: check if meter name is in use by other meters
-
+        cursor.execute(" SELECT name "
+                       " FROM tbl_meters "
+                       " WHERE name = %s ", (name,))
+        if cursor.fetchone() is not None:
+            cursor.close()
+            cnx.disconnect()
+            raise falcon.HTTPError(falcon.HTTP_404, title='API.BAD_REQUEST',
```
Comment (guangyuzhang) [inline meter.py]: status should be 400

File: meter.py [hunk 2]
```diff
@@ -168,18 +168,49 @@ def on_put(req, resp, id_):
...
+                                   description='API.METER_NAME_IS_ALREADY_IN_USE')
         # TODO 5: update meter in database
-
+        put_values = (" INSERT INTO tbl_meters "
+                      "    (name, uuid, description) "
+                      " VALUES (%s, %s, %s) ")
+        cursor.execute(put_values, (name,
+                                    str(uuid.uuid4()),
```
Comment (guangyuzhang) [inline meter.py]: don't change uuid

File: meter.py [hunk 3]
```diff
@@ -168,18 +168,49 @@ def on_put(req, resp, id_):
...
+                       " FROM tbl_meters "
+                       " WHERE name = %s ", (name,))
+        if cursor.fetchone() is not None:
+            cursor.close()
+            cnx.disconnect()
+            raise falcon.HTTPError(falcon.HTTP_404, title='API.BAD_REQUEST',
+                                   description='API.METER_NAME_IS_ALREADY_IN_USE')
         # TODO 5: update meter in database
-
+        put_values = (" INSERT INTO tbl_meters "
```
Comment (guangyuzhang) [inline meter.py]: use UPDATE instead of INSERT

File: meter.py [hunk 4]
```diff
@@ -168,18 +168,49 @@ def on_put(req, resp, id_):
...
+        if cursor.fetchone() is not None:
+            cursor.close()
+            cnx.disconnect()
+            raise falcon.HTTPError(falcon.HTTP_404, title='API.BAD_REQUEST',
+                                   description='API.METER_ID_IS_ALREADY_IN_USE')
         # TODO 4: check if meter name is in use by other meters
-
+        cursor.execute(" SELECT name "
+                       " FROM tbl_meters "
+                       " WHERE name = %s ", (name,))
```
Comment (guangyuzhang) [inline meter.py]: should add id !=_id to exclude the meter itself

**Comment by tianlinzhong** [comment]
更正

**Pull request closed without merging**